apps/passport-server/src/services/generic-issuance/pipelines/LemonadePipeline.ts

```diff
@@ -1941,6 +1969,377 @@ export class LemonadePipeline implements BasePipeline {
 
     return ids;
   }
+
+  /**
+   * Returns {@link MemberCriteria} for each of the ticket types this user has
+   * permission to check in.
+   */
+  private async ticketsUserCanCheckIn(
+    checkerEmail: string
+  ): Promise<MemberCriteria[]> {
+    if (this.definition.options.superuserEmails?.includes(checkerEmail)) {
+      // The user can check in all of the event/product combinations on this
+      // pipeline
+      return this.definition.options.events.flatMap((eventConfig) =>
+        eventConfig.ticketTypes.map((ticketType) => ({
+          eventId: eventConfig.genericIssuanceEventId,
+          productId: ticketType.genericIssuanceProductId
+        }))
+      );
+    }
+
+    // Get all of the products that the checker owns
+    const checkerProductIds: string[] = [];
+    for (const checkerTicketAtom of await this.db.loadByEmail(
+      this.id,
+      checkerEmail
+    )) {
+      checkerProductIds.push(checkerTicketAtom.productId);
+    }
+    for (const manualTicket of this.getManualTicketsForEmail(checkerEmail)) {
+      checkerProductIds.push(manualTicket.productId);
+    }
+
+    // Return any event/product combinations for which the user has a superuser
+    // ticket.
+    return this.definition.options.events.reduce((memo, eventConfig) => {
+      if (
+        // Does the user own a superuser product for this event?
+        eventConfig.ticketTypes.some(
+          (ticketType) =>
+            ticketType.isSuperUser &&
+            checkerProductIds.includes(ticketType.genericIssuanceProductId)
+        )
+      ) {
+        // In that case, they can check in any of the product types for this event
+        memo.push(
+          ...eventConfig.ticketTypes.map((ticketType) => ({
+            eventId: eventConfig.genericIssuanceEventId,
+            productId: ticketType.genericIssuanceProductId
+          }))
+        );
+      }
+      return memo;
+    }, [] as MemberCriteria[]);
+  }
+
+  /**
+   * Returns offline tickets that the user is entitled to check in, if any.
+   */
+  public async getOfflineTickets(
+    checkerEmail: string
+  ): Promise<PodboxOfflineTicket[]> {
+    return traced(LOG_NAME, "getOfflineTickets", async (span) => {
+      tracePipeline(this.definition);
+
+      if (this.definition.options.offlineCheckin === undefined) {
+        throw new PCDHTTPError(
+          401,
+          "Offline check-in is not enabled for this pipeline"
+        );
+      }
+
+      const checkerTicketCriteria =
+        await this.ticketsUserCanCheckIn(checkerEmail);
+      if (checkerTicketCriteria.length === 0) {
+        span?.setAttribute("offline_tickets_returned", 0);
+        // User can't check anything in, so they don't get any offline tickets.
+        return [];
+      }
+
+      const enabledFields = this.definition.options.offlineCheckin.fields;
```

Comment:
neat - cc @rrrliu, this enables the offline tickets we sync to the client to omit some PII based on the configuration of the pipeline